core/math/frustum.cpp

```diff
@@ -0,0 +1,249 @@
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "frustum.h"
+
+#include "core/math/projection.h"
+#include "core/math/transform_3d.h"
+#include "core/math/vector2.h"
+
+void Frustum::set_perspective(real_t p_fovy_degrees, real_t p_aspect, real_t p_z_near, real_t p_z_far, bool p_flip_fov) {
+	if (p_flip_fov) {
+		p_fovy_degrees = get_fovy(p_fovy_degrees, 1.0 / p_aspect);
+	}
+
+	// Inspired by https://iquilezles.org/articles/frustum/
+	real_t an = Math::deg_to_rad(p_fovy_degrees / 2);
+	real_t si = sin(an);
+	real_t co = cos(an);
+	real_t mag = Math::sqrt(co * co + si * si * p_aspect * p_aspect);
+
+	planes[Projection::PLANE_NEAR] = Plane(Vector3(0, 0, 1), -p_z_near);
+	planes[Projection::PLANE_FAR] = Plane(Vector3(0, 0, -1), p_z_far);
+	planes[Projection::PLANE_LEFT] = Plane(Vector3(-co, 0, si * p_aspect) / mag, 0);
+	planes[Projection::PLANE_TOP] = Plane(Vector3(0, co, si), 0);
+	planes[Projection::PLANE_RIGHT] = Plane(Vector3(co, 0, si * p_aspect) / mag, 0);
+	planes[Projection::PLANE_BOTTOM] = Plane(Vector3(0, -co, si), 0);
+}
+
+void Frustum::set_perspective(real_t p_fovy_degrees, real_t p_aspect, real_t p_z_near, real_t p_z_far, bool p_flip_fov, int p_eye, real_t p_intraocular_dist, real_t p_convergence_dist) {
+	if (p_flip_fov) {
+		p_fovy_degrees = get_fovy(p_fovy_degrees, 1.0 / p_aspect);
+	}
+
+	real_t left, right, modeltranslation, ymax, xmax, frustumshift;
```

Comment:
```suggestion
	real_t left, right, model_translation, y_max, x_max, frustum_shift;
```
Something like this here and below. Not sure about `y_max`, `x_max`, but it makes sense for `model_translation` and `frustum_shift`.